src/main/java/com/thebois/models/beings/Pawn.java

```diff
@@ -48,4 +59,63 @@ private void setRandomDestination() {
         setPath(newPath);
     }
 
+    protected void updateClosestStructure() {
+        if (closestStructure.isPresent() && getFinalDestination().isPresent()) {
+            // If the structure is completed or destination is unreachable pick a new one
+            if (closestStructure.get().isCompleted() || closestStructure.get().getPosition().equals(
+                getFinalDestination().get())) {
+                closestStructure = findIncompleteStructure();
+                setClosestStructureAsFinalDestination();
+            }
+        }
+        else {
+            closestStructure = findIncompleteStructure();
+            setClosestStructureAsFinalDestination();
+        }
+    }
+
+    protected Optional<IStructure> findIncompleteStructure() {
+        final Collection<Optional<IStructure>> structures =
+            getFinder().findNearestStructures(this.getPosition(), 10);
+
+        for (final Optional<IStructure> structure : structures) {
+            if (structure.isPresent()) {
+                if (!structure.get().isCompleted()) {
+                    return structure;
+                }
+            }
+            else {
+                return Optional.empty();
+            }
+        }
+        return Optional.empty();
+    }
+
+    private void setClosestStructureAsFinalDestination() {
+        closestStructure.ifPresent(structure -> setFinalDestination(structure.getPosition()));
+    }
+
+    private void setFinalDestination(final Position destination) {
+        final Position nextTo = nearestPositionNextTo(destination);
+        if (getFinalDestination().isPresent()) {
+            if (!getFinalDestination().get().equals(nextTo)) {
+                final Collection<Position> newPath = getPathFinder().path(getPosition(), nextTo);
+                setPath(newPath);
+            }
+        }
+        else {
+            final Collection<Position> newPath = getPathFinder().path(getPosition(), nextTo);
+            setPath(newPath);
+        }
+    }
+
+    protected void deliverItemToStructure(final Optional<IStructure> structure) {
```

Comment:
style

How come there are `protected` methods in `Pawn`? I can't imagine any subtype of `Pawn` that would need access to these methods, so they should probably be private.